app/services/visit_request/final_confirmation.rb

```diff
@@ -3,35 +3,20 @@ class FinalConfirmation < ApplicationService
     YES = 'yes'
     NO  = 'no'
 
```

Comment:
Think all should be as it was. Just add `visit_request` method here.
```ruby
def visit_request
   @visit_request ||= VisitRequest.find_by(token: params[:token])
end
```

Reply:
Currently I see validations in this file useless. In case we leave them, service has to return some validation errors (not `nil`) and controller has to deal with them and present appropriate flash message. Thus, it requires more complex refactoring, imho.